web/crux/src/app/image/image.http.controller.ts

```diff
@@ -1,15 +1,19 @@
-import { Controller, Post, Body, UseGuards, UseInterceptors } from '@nestjs/common'
-import { ApiBody, ApiCreatedResponse } from '@nestjs/swagger'
+import { Controller, Post, Body, UseGuards, UseInterceptors, Patch, UseFilters } from '@nestjs/common'
+import { ApiBody, ApiCreatedResponse, ApiOkResponse } from '@nestjs/swagger'
 import { AuditLogLevel } from 'src/decorators/audit-logger.decorators'
-import { AddImagesToVersionRequest, ImageListResponse } from 'src/grpc/protobuf/proto/crux'
+import HttpExceptionFilter from 'src/filters/http-exception.filter'
+import { Empty } from 'src/grpc/protobuf/proto/common'
+import { AddImagesToVersionRequest, ImageListResponse, PatchImageRequest } from 'src/grpc/protobuf/proto/crux'
 import HttpLoggerInterceptor from 'src/interceptors/http.logger.interceptor'
-import { AddImagesToVersionRequestDto, ImageListResponseDto } from 'src/swagger/crux.dto'
+import { AddImagesToVersionRequestDto, ImageListResponseDto, PatchImageRequestDto } from 'src/swagger/crux.dto'
 import JwtAuthGuard from '../token/jwt-auth.guard'
 import ImageService from './image.service'
+import ImagePatchValidationPipe from './pipes/image.patch.pipe'
 
 @Controller('image')
 @UseGuards(JwtAuthGuard)
 @UseInterceptors(HttpLoggerInterceptor)
+@UseFilters(HttpExceptionFilter)
```

Comment:
Can we put this globally onto the http service itself in the main.ts?

Reply:
Nope, unused if we are calling the service via gRPC.

Reply:
Ah right. I thought nest's http framework is added by some function like the grpc's connectMicroservice() call, but i can see now that is not the case at all, and the app itself is the http framework.